webapp/components/post_view/components/post_list.jsx

```diff
@@ -50,12 +54,96 @@ export default class PostList extends React.Component {
             this.introText = this.getArchivesIntroMessage();
         }
 
+        let lastViewed = Number.MAX_VALUE;
+        const member = ChannelStore.getMember(props.channel.id);
+        if (member != null) {
+            lastViewed = member.last_viewed_at;
+        }
+
         this.state = {
             isScrolling: false,
-            topPostId: null
+            topPostId: null,
+            lastViewed,
+            ownNewMessage: false
         };
     }
 
+    componentWillReceiveProps(nextProps) {
+        if (this.props.channel !== nextProps.channel) {
+            let lastViewed = Number.MAX_VALUE;
+            const member = ChannelStore.getMember(nextProps.channel.id);
+            if (member != null) {
+                lastViewed = member.last_viewed_at;
+            }
+            this.setState({
+                lastViewed,
+                ownNewMessage: false
+            });
+        }
+    }
+
+    setUnreadPost(post) {
```

Comment:
What do you mean by "any setting of data"? 
Aout this function - it's exactly was id `post_view_controller.jsx`  and `lastViewed` was a prop, but i've moved it down, because I can't reassign value of component prop from itself, so I make it a state value instead